app/controllers/users/reset_passwords_controller.rb

```diff
@@ -62,7 +79,7 @@ def build_user
     end
 
     def handle_successful_password_reset
-      resource.update(password: user_params[:password])
+      resource.update(password: user_params[:password], confirmed_at: Time.current)
```

Comment:
It wasn't. Previously, confirmation could only happen via the confirmation email link. See my commit message, and these 2 comments for more details:

https://github.com/18F/identity-private/issues/1151#issuecomment-267510684
https://github.com/18F/identity-private/issues/1151#issuecomment-267519136

However, you reminded me that I meant to make this conditional so that it only sets `confirmed_at` if the user is not yet confirmed. That way, `confirmed_at` won't get updated every time a user resets their password. And we might also want to capture this scenario in analytics to see how many unconfirmed users try to reset their password.